php-frontend/src/main/java/org/sonar/php/tree/impl/statement/ForStatementTreeImpl.java

```diff
@@ -0,0 +1,239 @@
+      InternalSyntaxToken endForToken, InternalSyntaxToken eosToken
+  ) {
+    this(Kind.ALTERNATIVE_FOR_STATEMENT, header, colonToken, statements, endForToken, eosToken);
+  }
+
+  public ForStatementTreeImpl(ForStatementHeader header, StatementTree statement) {
+    this(Kind.FOR_STATEMENT, header, null, Collections.singletonList(statement), null,  null);
+  }
+
+  @Override
+  public SyntaxToken forToken() {
+    return header.forToken();
+  }
+
+  @Override
+  public SyntaxToken openParenthesisToken() {
+    return header.openParenthesisToken();
+  }
+
+  @Override
+  public SeparatedListImpl<ExpressionTree> init() {
+    return header.init();
+  }
+
+  @Override
+  public SyntaxToken firstSemicolonToken() {
+    return header.firstSemicolonToken();
+  }
+
+  @Override
+  public SeparatedListImpl<ExpressionTree> condition() {
+    return header.condition();
+  }
+
+  @Override
+  public SyntaxToken secondSemicolonToken() {
+    return header.secondSemicolonToken();
+  }
+
+  @Override
+  public SeparatedListImpl<ExpressionTree> update() {
+    return header.update();
+  }
+
+  @Override
+  public SyntaxToken closeParenthesisToken() {
+    return header.closeParenthesisToken();
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken colonToken() {
+    return colonToken;
+  }
+
+  @Override
+  public List<StatementTree> statements() {
+    return statements;
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken endforToken() {
+    return endForToken;
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken eosToken() {
+    return eosToken;
+  }
+
+  @Override
+  public Kind getKind() {
+    return KIND;
+  }
+
+  @Override
+  public Iterator<Tree> childrenIterator() {
+    return Iterators.concat(
+        Iterators.forArray(forToken(), openParenthesisToken()),
+        init().elementsAndSeparators(),
+        Iterators.singletonIterator(firstSemicolonToken()),
+        condition().elementsAndSeparators(),
+        Iterators.singletonIterator(secondSemicolonToken()),
+        update().elementsAndSeparators(),
+        Iterators.forArray(closeParenthesisToken(), colonToken),
+        statements.iterator(),
+        Iterators.forArray(endForToken, eosToken));
+  }
+
+  @Override
+  public void accept(VisitorCheck visitor) {
+    visitor.visitForStatement(this);
+  }
+
+  /**
+   * Utility class hidden from API (it's mainly created to avoid duplication in grammar)
+   */
+  public static class ForStatementHeader implements Tree {
+
+    private final InternalSyntaxToken forToken;
+    private final InternalSyntaxToken openParenthesisToken;
+    private final SeparatedListImpl<ExpressionTree> init;
+    private final InternalSyntaxToken firstSemicolonToken;
+    private final SeparatedListImpl<ExpressionTree> condition;
+    private final InternalSyntaxToken secondSemicolonToken;
+    private final SeparatedListImpl<ExpressionTree> update;
+    private final InternalSyntaxToken closeParenthesisToken;
+
+    public ForStatementHeader(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)